### PR TITLE
Separate effect outputs v2

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #3729

This redraft of #3735 avoids using std::any and also adds more const-correctness of arguments.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by Paul-Licameli** [comment]
Latest force-push only changed commit comments.

**Comment by Paul-Licameli** [comment]
Another push fixes one foolish mistake where I called `reserve` but meant `resize` in `LV2Effect::MakeOutputs`

File: src/effects/ladspa/LadspaEffect.cpp [hunk 1]
```diff
@@ -1639,8 +1648,15 @@ LADSPA_Handle LadspaEffect::InitInstance(
...
       LADSPA_PortDescriptor d = mData->PortDescriptors[p];
-      if (LADSPA_IS_PORT_CONTROL(d))
-         mData->connect_port(handle, p, &controls[p]);
+      if (LADSPA_IS_PORT_CONTROL(d)) {
+         if (LADSPA_IS_PORT_INPUT(d))
+            mData->connect_port(handle, p, &controls[p]);
+         else {
+            static LADSPA_Data sink;
+            mData->connect_port(handle, p,
+               pOutputs ? &pOutputs->controls[p] : &sink);
```
Comment (vsverchinsky) [inline src/effects/ladspa/LadspaEffect.cpp:1659]: Correct me if I'm wrong, but effect writes `LadspaEffectOutputs` on the processing thread, here you bind control directly to the location of these outputs, what prevents data races in that case?

Comment (Paul-Licameli) [reply src/effects/ladspa/LadspaEffect.cpp:1659]: You are right.  I explained the bug I introduce knowingly here but then correct in the second PR.

The two should be merged as a whole -- or perhapsw the whole sequence of four should be -- but I thought the first PR was big enough already.

**Pull request merged**